internal/addrs/instance_key.go

```diff
@@ -53,6 +53,11 @@ func ParseInstanceKey(key cty.Value) (InstanceKey, error) {
 // of a configuration object that does not use "count" or "for_each" at all.
 var NoKey InstanceKey
 
+// WildcardKey represents the "unknown" value of an InstanceKey. This is used
+// within the deferral logic to express absolute module and resource addresses
+// that are not known at the time of planning.
+var WildcardKey InstanceKey = StringKey("*")
```

Comment:
While unlikely, `"*"` is a valid instance key on its own. Is it possible that could collide here?

Reply:
This would only be set when combined with the `DeferredReasonInstanceCountUnknown` reason. That reason will never have a "valid" instance key associated with it. I've used this just as a convenience for when the address is rendered, in reality nothing should attempt to parse an address from a deferred change with the mentioned reason as the value in there is inherently meaningless.